src/main/java/io/cdap/plugin/hive/action/HiveExportConfig.java

```diff
@@ -40,4 +65,135 @@ public HiveExportConfig(String connectionString, String user, String password, S
     this.path = path;
     this.overwrite = overwrite;
   }
+
+  private HiveExportConfig(Builder builder) {
+    super(builder.connectionString, builder.user, builder.password, builder.statement);
+    delimiter = builder.delimiter;
+    path = builder.path;
+    overwrite = builder.overwrite;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(HiveExportConfig copy) {
+    return builder()
+      .setConnectionString(copy.getConnectionString())
+      .setUser(copy.getUser())
+      .setPassword(copy.getPassword())
+      .setStatement(copy.getStatement())
+      .setDelimiter(copy.delimiter)
+      .setPath(copy.path)
+      .setOverwrite(copy.overwrite);
+  }
+
+  @Nullable
+  public String getDelimiter() {
+    return delimiter;
+  }
+
+  public String getPath() {
+    return path;
+  }
+
+  @Nullable
+  public String getOverwrite() {
+    return overwrite;
+  }
+
+  @Override
+  public void validate(FailureCollector failureCollector) {
+    super.validate(failureCollector);
+
+    if (!containsMacro(STATEMENT)) {
+      // validate hive command. For export we only accept Select statements
+      SqlParser parser = SqlParser.create(getStatement());
+      try {
+        SqlNode sqlNode = parser.parseQuery();
+        if (!(sqlNode instanceof SqlSelect)) {
+          failureCollector.addFailure("Hive Export only uses Select statements.",
+                                      "Provide valid hive select statement.")
+            .withConfigProperty(STATEMENT);
+        }
+      } catch (SqlParseException e) {
+        failureCollector.addFailure("Error while parsing select statement.",
```

Comment:
I think we should add e.getMessage() to the message.
Also I think corrective action does not provide any extra value, though can be removed.

Reply:
Done.